MapboxNavigation/NavigationMapView.swift

```diff
@@ -1209,6 +1209,24 @@ public protocol NavigationMapViewDelegate: class {
     @objc(navigationMapView:shapeDescribingWaypoints:)
     optional func navigationMapView(_ mapView: NavigationMapView, shapeFor waypoints: [Waypoint]) -> MGLShape?
     
+    /**
+     Asks the receiver to return an MGLAnnotationImage that describes the image used an annotation.
+     - parameter mapView: The MGLMapView.
+     - parameter annotation: The annotation to be styled.
+     - returns: Optionally, a `MGLAnnotationImage` that defines the image used for the annotation.
+     */
+    @objc(navigationMapView:imageForAnnotation:)
+    optional func navigationMapView(_ mapView: MGLMapView, imageFor annotation: MGLAnnotation) -> MGLAnnotationImage?
```

Comment:
FYI, this implements part of #685 but takes us further away from being able to allow the developer to just use any MGLMapViewDelegate method, as suggested in https://github.com/mapbox/mapbox-navigation-ios/issues/685#issuecomment-334294064.

Reply:
Yeah I agree, however lets make the breaking change for all methods in a single pr.